trtllm-truss/model/model.py

```diff
@@ -0,0 +1,74 @@
+import numpy as np
+from client import UserData, TritonClient
+from threading import Thread
+from utils import prepare_grpc_tensor
+from pathlib import Path
+from itertools import count
+
+TRITON_MODEL_REPOSITORY_PATH = Path("/packages/inflight_batcher_llm/")
+
+class Model:
+    def __init__(self, **kwargs):
+        self._data_dir = kwargs["data_dir"]
+        self._config = kwargs["config"]
+        self._secrets = kwargs["secrets"]
+        self._tensor_parallel_count = None
+        self._request_id_counter = count(start=1)
+        self.triton_client = None
+
+    def load(self):
+        self._tensor_parallel_count = self._config["model_metadata"].get("tensor_parallelism", 1)
+        self.triton_client = TritonClient(self._data_dir, TRITON_MODEL_REPOSITORY_PATH, self._tensor_parallel_count)
+        self.triton_client.load_server_and_model(
+            env={
+                "HUGGING_FACE_HUB_TOKEN": self._secrets["hf_access_token"]
+                "triton_tokenizer_repository": self._config["model_metadata"]["tokenizer_repository"],
+            }
+        )
+
+    def predict(self, model_input):
```

Comment:
We should try to use async predict. Sync predict runs on a thread pool which has limited number of threads and can limit concurrency. Plus creating a new thread per request is not ideal. cc @squidarth who may know of examples of where we use async predict.

Reply:
Ah, I missed the yield before. So this predict function is a generator, right?

Reply:
it's not obvious to me that there would be a big perf increase by switching to having this by async (it's true that doing things this way will produce another thread). It's a medium lift at least to switch it, since we'd have to change the TritonClient implementation to also be async